File: app-shell-ui/src/lib/interface/Project.ts
export interface ProjectDto {
    id: string;
    name: string;
    createdBy: string;
    bpm: number;
    quantizeDivision: string;
    masterVolume: number;
    masterMuted: boolean;
    createdAt: string;
    updatedAt: string;
}

export interface CreateProjectRequest {
    name: string;
    bpm: number;
    quantizeDivision: string;
    masterVolume?: number;
    masterMuted?: boolean;
}

export interface UpdateProjectRequest {
    name?: string;
    bpm?: number;
    quantizeDivision?: string;
    masterVolume?: number;
    masterMuted?: boolean
}

File: projects-api/src/main/kotlin/mike/dev/project_api/model/Project.kt
package mike.dev.project_api.model

import jakarta.persistence.*
import java.math.BigDecimal
import java.time.LocalDateTime
import java.util.UUID

@Entity
@Table(name = "projects")
class Project(
    @Column(nullable = false)
    var name: String,

    @Column(nullable = false)
    var createdBy: String,

    @Column(nullable = false)
    var bpm: Int,

    @Column(nullable = false)
    var quantizeDivision: String,

    @Column(nullable = false)
    var masterVolume: BigDecimal,

    @Column(nullable = false)
    var masterMuted: Boolean = false,

    @OneToMany(mappedBy = "project", cascade = [CascadeType.ALL], orphanRemoval = true)
    var channels: MutableList<Channel> = mutableListOf(),

    @OneToMany(mappedBy = "project", cascade = [CascadeType.ALL], orphanRemoval = true)
    var effects: MutableList<Effect> = mutableListOf(),

    @Column(nullable = false)
    var createdAt: LocalDateTime = LocalDateTime.now(),

    @Column(nullable = false)
    var updatedAt: LocalDateTime = LocalDateTime.now()
) {
    constructor():
        this(
                "New Project",
                "Anonymous",
                120,
                "4/4",
                BigDecimal.ONE,
                false,
                mutableListOf<Channel>(),
                mutableListOf<Effect>(),
                LocalDateTime.now(),
                LocalDateTime.now()
                )
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long = 0
}